code/modules/projectiles/projectile/beams.dm

```diff
@@ -28,6 +28,13 @@
 	wound_bonus = -100
 	bare_wound_bonus = -100
 
+/obj/projectile/beam/laser/prehit_pierce(atom/A)
```

Comment:
Isn't it more processor efficient to do it the way Loyalty does?

Reply:
The way loyalty does it is questionable.

Reply:
Not really, this always checks 1 variable, sets one variable, then if it hits something it checks one flag, and if it pierces it checks another flag and returns Loyalty just checks one flag whether it hits or misses, then sets one flag if it hits. It's less lines of code and less processor intensive.

Reply:
You'd like to think so, but I really don't see anything good about the way you did it.